android/app/src/main/java/com/microsoft/codepush/react/CodePushTelemetryManager.java

```diff
@@ -24,34 +34,51 @@ public CodePushTelemetryManager(Context applicationContext, String codePushPrefe
     public WritableMap getBinaryUpdateReport(String appVersion) {
         String previousStatusReportIdentifier = this.getPreviousStatusReportIdentifier();
         if (previousStatusReportIdentifier == null) {
-            this.recordDeploymentStatusReported(appVersion);
+            this.clearRetryStatusReport();
             WritableNativeMap reportMap = new WritableNativeMap();
-            reportMap.putString("appVersion", appVersion);
+            reportMap.putString(APP_VERSION_KEY, appVersion);
             return reportMap;
         } else if (!previousStatusReportIdentifier.equals(appVersion)) {
-            this.recordDeploymentStatusReported(appVersion);
+            this.clearRetryStatusReport();
             WritableNativeMap reportMap = new WritableNativeMap();
             if (this.isStatusReportIdentifierCodePushLabel(previousStatusReportIdentifier)) {
                 String previousDeploymentKey = this.getDeploymentKeyFromStatusReportIdentifier(previousStatusReportIdentifier);
                 String previousLabel = this.getVersionLabelFromStatusReportIdentifier(previousStatusReportIdentifier);
-                reportMap.putString("appVersion", appVersion);
-                reportMap.putString("previousDeploymentKey", previousDeploymentKey);
-                reportMap.putString("previousLabelOrAppVersion", previousLabel);
+                reportMap.putString(APP_VERSION_KEY, appVersion);
+                reportMap.putString(PREVIOUS_DEPLOYMENT_KEY_KEY, previousDeploymentKey);
+                reportMap.putString(PREVIOUS_LABEL_OR_APP_VERSION_KEY, previousLabel);
             } else {
                 // Previous status report was with a binary app version.
-                reportMap.putString("appVersion", appVersion);
-                reportMap.putString("previousLabelOrAppVersion", previousStatusReportIdentifier);
+                reportMap.putString(APP_VERSION_KEY, appVersion);
+                reportMap.putString(PREVIOUS_LABEL_OR_APP_VERSION_KEY, previousStatusReportIdentifier);
             }
             return reportMap;
         }
 
         return null;
```

Comment:
initialize the reportMap outside of the above `if..else if..` block and then return it here. Remove intermediate returns in the `if...else if...` blocks

```
WriteableNativeMap reportMap = null;

if (...) {...}
else if (...) {...}

return reportMap;
```